qcodes/instrument_drivers/Galil/dmc_41x3.py

```diff
@@ -0,0 +1,964 @@
+        c.begin()
+        c.wait_till_motor_motion_complete()
+
+    def _setup_motion(self,
+                      rel_vec: np.ndarray,
+                      d: float,
+                      speed: float) -> None:
+
+        pos = self.controller.absolute_position()
+
+        a = int(np.floor(rel_vec[0] * d))
+        b = int(np.floor(rel_vec[1] * d))
+        c = int(np.floor(rel_vec[2] * d))
+
+        self._target = np.array([pos["A"] + a, pos["B"] + b, pos["C"] + c, 1])
+
+        if np.dot(self._plane_eqn, self._target) < 0:
+            raise RuntimeError(
+                f"Cannot move to {self._target[:3]}. Target location is below "
+                f"chip plane."
+            )
+
+        sp_a = int(np.floor(abs(rel_vec[0]) * speed))
+        if sp_a % 2 != 0:
+            sp_a += 1
+
+        sp_b = int(np.floor(abs(rel_vec[1]) * speed))
+        if sp_b % 2 != 0:
+            sp_b += 1
+
+        sp_c = int(np.floor(abs(rel_vec[2]) * speed))
+        if sp_c % 2 != 0:
+            sp_c += 1
+
+        motor_a = self.controller.motor_a
+        motor_b = self.controller.motor_b
+        motor_c = self.controller.motor_c
+
+        motor_a.relative_position(a)
+        motor_a.speed(sp_a)
+        motor_a.acceleration(self.acceleration)
+        motor_a.deceleration(self.deceleration)
+        motor_a.servo_here()
+
+        motor_b.relative_position(b)
+        motor_b.speed(sp_b)
+        motor_b.acceleration(self.acceleration)
+        motor_b.deceleration(self.deceleration)
+        motor_b.servo_here()
+
+        motor_c.relative_position(c)
+        motor_c.speed(sp_c)
+        motor_c.acceleration(self.acceleration)
+        motor_c.deceleration(self.deceleration)
+        motor_c.servo_here()
+
+    def _move(self) -> None:
+        self.controller.begin_motors()
+        self.controller.wait_till_motion_complete()
+
+    def _pick_up(self) -> None:
+
+        self._setup_motion(rel_vec=self._n,
+                           d=self._arm_pick_up_dis,
+                           speed=self.speed)
+        self._move()
+
+    def _put_down(self) -> None:
+
+        motion_vec = -1 * self._n
+
+        pos = self.controller.absolute_position()
+        x1 = pos["A"]
+        y1 = pos["B"]
+        z1 = pos["C"]
+        current = np.array([x1, y1, z1, 1])
+
+        denominator = \
+            np.sqrt(
+                self._plane_eqn[0] ** 2 + \
+                self._plane_eqn[1] ** 2 + \
+                self._plane_eqn[2] ** 2
+            )
+        d = abs(sum(self._plane_eqn * current)) / denominator
+
+        self._setup_motion(rel_vec=motion_vec, d=d, speed=self.speed)
+        self._move()
+
+    def move_towards_left_bottom_position(self) -> None:
+
+        self._pick_up()
+
+        motion_vec = -1 * self._a
+        self._setup_motion(rel_vec=motion_vec, d=self.norm_a, speed=self.speed)
+        self._move()
+        self.current_row = 1
+        self.current_pad = 1
+
+        self._put_down()
```

Comment:
it's not critical but (in all methods that change row/pad) i'd recommend to set row/pad last so that it's ensured that the motions were successfully executed. otherwise if put down fails, current row and pad are also changed which is pre-mature :)
```suggestion
        self._put_down()

        self.current_row = 1
        self.current_pad = 1
```